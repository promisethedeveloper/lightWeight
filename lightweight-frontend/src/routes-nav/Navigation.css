@import url("https://fonts.googleapis.com/css2?family=Open+Sans:wght@700&display=swap");

.Navigation {
	background-color: white;
	border-bottom: 2px solid rgba(0, 0, 0, 0.125);
}

.nav-link.active {
	font-weight: bold;
	color: black;
}

h4 {
	font-weight: 600;
	font-family: "Open Sans", sans-serif;
}

.Navigation-brand {
	color: #040303;
	font-weight: 700;
}

.nav-link {
	color: #040303;
}
