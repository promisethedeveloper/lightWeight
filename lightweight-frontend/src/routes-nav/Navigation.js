import React, { useContext } from "react";
import { Link, NavLink } from "react-router-dom";
import UserContext from "../auth/UserContext";
import Container from "react-bootstrap/Container";
import Navbar from "react-bootstrap/Navbar";
import "./Navigation.css";

/** Navigation bar for the site. Shows up on every page
 *
 * When User is logged in, show links to the main areas of the site.
 * When User is not logged in, show links to Login and Signup forms.
 *
 */

const Navigation = ({ logout }) => {
	const { currentUser } = useContext(UserContext);

	const loggedInNav = () => {
		return (
			<ul className="navbar-nav ml-auto">
				<li className="nav-item mr-4">
					<NavLink className="nav-link" to="/workouts">
						Workouts
					</NavLink>
				</li>
				<li className="nav-item mr-4">
					<NavLink className="nav-link" to="/profile">
						Profile
					</NavLink>
				</li>
				<li className="nav-item">
					<Link className="nav-link" to="/" onClick={logout}>
						Log Out {currentUser.first_name || currentUser.username}
					</Link>
				</li>
			</ul>
		);
	};

	const loggedOutNav = () => {
		return (
			<ul className="navbar-nav ml-auto Navigation-logout">
				<li className="nav-item mr-4">
					<NavLink className="nav-link" to="/login">
						Login
					</NavLink>
				</li>
				<li className="nav-item mr-4">
					<NavLink className="nav-link" to="/signup">
						Sign Up
					</NavLink>
				</li>
			</ul>
		);
	};

	return (
		<Navbar collapseOnSelect expand="lg">
			<Container>
				<div className="container-fluid">
					<Navbar.Brand>
						<Link className="navbar-brand" to="/">
							<h4 className="Navigation-brand">LIGHT WEIGHT</h4>
						</Link>
					</Navbar.Brand>
					<Navbar.Toggle aria-controls="responsive-navbar-nav" />
					<Navbar.Collapse id="responsive-navbar-nav">
						{currentUser ? loggedInNav() : loggedOutNav()}
					</Navbar.Collapse>
				</div>
			</Container>
		</Navbar>
	);
};

export default Navigation;
