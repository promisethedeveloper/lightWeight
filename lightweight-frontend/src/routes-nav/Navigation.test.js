import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UserContext from "../auth/UserContext";
import Navigation from "./Navigation";

/** Smoke test */
it("renders without crashing", () => {
	const currentUser = { username: "John" };
	render(
		<MemoryRouter>
			<UserContext.Provider value={{ currentUser }}>
				<Navigation />
			</UserContext.Provider>
		</MemoryRouter>
	);
});

/** Snapshot test */
it("Should match snapshots", () => {
	const currentUser = { username: "John" };
	const { asFragment } = render(
		<MemoryRouter>
			<UserContext.Provider value={{ currentUser }}>
				<Navigation />
			</UserContext.Provider>
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("renders a Workout, Profile, and Log Out naviagtion bar if there is a current user", () => {
	const currentUser = { username: "John" };
	const { getByText } = render(
		<MemoryRouter>
			<UserContext.Provider value={{ currentUser }}>
				<Navigation />
			</UserContext.Provider>
		</MemoryRouter>
	);
	expect(getByText("Workouts")).toBeInTheDocument();
	expect(getByText("Profile")).toBeInTheDocument();
	expect(getByText("Log Out John")).toBeInTheDocument();
});

it("renders Login and Sign Up buttons if there is no current user", () => {
	const { getByText } = render(
		<MemoryRouter>
			<UserContext.Provider value={{}}>
				<Navigation />
			</UserContext.Provider>
		</MemoryRouter>
	);
	expect(getByText("Login")).toBeInTheDocument();
	expect(getByText("Sign Up")).toBeInTheDocument();
});
