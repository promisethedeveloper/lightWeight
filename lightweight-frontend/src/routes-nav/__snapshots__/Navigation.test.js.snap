// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshots 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-light"
  >
    <div
      class="container"
    >
      <div
        class="container-fluid"
      >
        <span
          class="navbar-brand"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            <h4
              class="Navigation-brand"
            >
              LIGHT WEIGHT
            </h4>
          </a>
        </span>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <ul
            class="navbar-nav ml-auto"
          >
            <li
              class="nav-item mr-4"
            >
              <a
                class="nav-link"
                href="/workouts"
              >
                Workouts
              </a>
            </li>
            <li
              class="nav-item mr-4"
            >
              <a
                class="nav-link"
                href="/profile"
              >
                Profile
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/"
              >
                Log Out John
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
